### PR TITLE
Chore: Move and wrap Cascader component to @grafana/ui

### LINKED ISSUE #19042: Move & wrap Cascader component out of Explore and into @grafana/ui
Currently we are using the rc-cascader component directly in explore and overriding css in explore.css.

Should move this component out of explore and into @grafana/ui and add it to storybook. It's a really useful component that many data source query editors (and other areas) could use.

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Moves and wraps rc-cascader component to @grafana/ui
Also adds type information.

**Which issue(s) this PR fixes**:
Closes #19042

### CODE REVIEW

File: packages/grafana-ui/package.json [hunk 1]
```diff
@@ -39,8 +39,10 @@
     "lodash": "4.17.15",
     "moment": "2.24.0",
     "papaparse": "4.6.3",
+    "rc-cascader": "^0.17.5",
```
Comment (dprokop) [inline packages/grafana-ui/package.json]:
```suggestion
    "rc-cascader": "0.17.5",
```

File: packages/grafana-ui/package.json [hunk 2]
```diff
@@ -39,8 +39,10 @@
...
     "rc-drawer": "3.0.2",
     "rc-time-picker": "^3.7.2",
+    "rc-trigger": "^4.0.0-alpha.5",
```
Comment (dprokop) [inline packages/grafana-ui/package.json]:
```suggestion
    "rc-trigger": "4.0.0-alpha.5",
```

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+// @ts-ignore
+import RCCascader from 'rc-cascader';
+import { BuildInPlacements as BuiltInPlacements } from 'rc-trigger';
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Cascader/Cascader.tsx]: Not super wild about adding rc-trigger only to get this one type. It's yet another package that we need to manage(which we are kind of failing at atm). Can we redefine the type instead?

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+export interface CascaderOption {
+  label: string;
+  value: string;
+
+  children?: CascaderOption[];
+  disabled?: boolean;
+}
+
+export interface CascaderProps {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Cascader/Cascader.tsx]: Would be great to limit this API only to the props that we are using. Don't see much value in wrapping a component and exposing its entire API. WDYT?

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx (hunk 3 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/Cascader/Cascader.tsx]: rc-cascader already has rc-trigger as a dependency so adding it is really just to expose the `BuiltInPlacements` type

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx (hunk 4 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/Cascader/Cascader.tsx]: Sure, makes sense to me

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx (hunk 3 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/Cascader/Cascader.tsx]: looked into this a bit more and it seems like the version of rc-trigger that rc-cascader depends on doesn't have types, so you're right, I'll just have to redefine the type.

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx (hunk 4 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Cascader/Cascader.tsx]: yea, agree. makes upgrades easier and less risk of breaking plugins that use this :)

**Review by torkelo — COMMENTED** [review]
Great initiative in moving this to @grafana/ui , could be very useful for external plugins!

File: packages/grafana-ui/src/components/Cascader/Cascader.story.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    disabled: boolean('Disabled', false),
+    text: text('Button Text', 'Click me!'),
+    options: object('Options', [
+      { label: 'A', value: 'A', children: [{ label: 'B', value: 'B' }, { label: 'C', value: 'C' }] },
+      { label: 'D', value: 'D' },
+    ]),
+  };
+};
+
+const CascaderStories = storiesOf('UI/Cascader', module);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Cascader/Cascader.story.tsx]: @kaydelaney FYI you don't need to use `storiesOf` anymore - you can use https://storybook.js.org/docs/formats/component-story-format/ instead

**Review by dprokop — APPROVED** [review]
One small comment about stories format left. Apart from that I think this is good :)

**Pull request merged**